### PR TITLE
Added debug render modes, wireframe rendering

### PR DESCRIPTION
**Pull request by slimbuck**
This PR:
- updates wireframe view to render over mesh solids and adds wireframe color
- adds render mode option to display various debug material components
- update to latest engine and pcui

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -88,5 +87,8 @@
     "build:local": "cross-env ENGINE_PATH=../engine npm run build",
     "watch:local": "cross-env ENGINE_PATH=../engine npm run watch",
     "lint": "eslint --ignore-pattern 'src/lib' --ext .ts src"
+  },
+  "dependencies": {
+    "fflate": "^0.7.4"
```
Comment (willeastcott) [inline package.json:92]: I suspect this is a devDependency

File: rollup.config.mjs [hunk 2]
```diff
@@ -8,31 +8,37 @@ import resolve from "@rollup/plugin-node-resolve";
...
+const EXTRAS_DIR = path.resolve(ENGINE_DIR, 'build', 'playcanvas-extras.mjs');
+const PCUI_DIR = path.resolve(process.env.PCUI_PATH || 'node_modules/@playcanvas/pcui', 'react');
 
-const ENGINE_NAME = PROD_BUILD ? 'playcanvas.mjs' : 'playcanvas.dbg.mjs';
+const ENGINE_NAME = (BUILD_TYPE === 'debug') ? 'playcanvas.dbg.mjs' : 'playcanvas.mjs';
 const ENGINE_PATH = path.resolve(ENGINE_DIR, 'build', ENGINE_NAME);
-const EXTRAS_PATH = path.resolve(ENGINE_DIR, 'build', 'playcanvas-extras.js');
-const PCUI_PATH = path.resolve(PCUI_DIR, 'react');
+
+console.log(EXTRAS_DIR);
```
Comment (willeastcott) [inline rollup.config.mjs:27]: Delete?

File: rollup.config.mjs [hunk 3]
```diff
@@ -81,13 +87,10 @@ export default {
...
         typescript({
-            tsconfig: 'tsconfig.json',
-            tsconfigDefaults: { compilerOptions: tsCompilerOptions },
-            clean: true
+            compilerOptions: tsCompilerOptions
         }),
         json(),
-        (PROD_BUILD && terser()),
-        // visualizer()
+        (BUILD_TYPE !== 'debug') && terser(),
```
Comment (willeastcott) [inline rollup.config.mjs:93]: Remove trailing comma

File: package.json (hunk 1 above)
Comment (slimbuck) [reply package.json:92]: yes! thanks!

**Review by mvaligursky — APPROVED** [review]

File: rollup.config.mjs (hunk 3 above)
Comment (slimbuck) [reply rollup.config.mjs]: done!

File: rollup.config.mjs (hunk 2 above)
Comment (slimbuck) [reply rollup.config.mjs]: yes!

**Review by willeastcott — APPROVED** [review]

**Comment by willeastcott** [comment]
Wow - really awesome improvements here. 🙌

File: src/index.tsx [hunk 4]
```diff
@@ -28,6 +29,44 @@ declare global {
...
+#ifdef DEBUG_METALNESS_PASS
+gl_FragColor = vec4(vec3(litShaderArgs.metalness) , 1.0);
+#endif
+
+#ifdef DEBUG_AO_PASS
+gl_FragColor = vec4(gammaCorrectOutput(vec3(litShaderArgs.ao)), 1.0);
+#endif
+
+#ifdef DEBUG_EMISSION_PASS
+gl_FragColor = vec4(litShaderArgs.emission, 1.0);
```
Comment (mvaligursky) [inline src/index.tsx:66]: needs gamma here too

**Pull request merged**